action.yml

```diff
@@ -5,15 +5,87 @@ branding:
   icon: tag
   color: yellow
 inputs:
-  gh-token:
-    description: GitHub token with repo access
-    required: true
+  push-tag:
+    description: 'Push the tag to the origin repository'
+    required: false
+    default: true # TODO: default false in v1?
+  push-major-version-branch:
+    description: 'Push to a branch matching the major version number on the origin repository'
+    required: false
+    default: true # TODO: default false in v1?
+  v-prefix:
+    description: "Whether to prefix the tag with the letter 'v'."
+    required: false
+    default: true # TODO: default false in v1?
+  create-release:
+    description: "Whether to create a release from the tag or not. 'true', 'false'." # TODO: Support 'draft'
+    required: false
+    default: 'true' # TODO: default false in v1?
+  workdir:
+    description: 'Directory with the code to tag'
+    required: false
+    default: '.'
+
+outputs:
+  tag:
+    description: 'The tag that was created'
+    value: ${{ steps.tag.outputs.VERSION_TAG }}
 runs:
   using: composite
   steps:
     - uses: actions/checkout@v4
-    - name: Tag & Release
+    - id: tag
+      name: Tag
       shell: bash
-      run: sh ${{ github.action_path }}/src/tag-release.sh
+      working-directory: ${{ inputs.workdir }}
+      run: |
+        curl -sL https://git.io/autotag-install | sh --
+        # fetch all tags and history:
+        git fetch --tags --unshallow --prune
+
+        # This probably needs to be addressed in v1 for projects not using 'main'
+        if [ "$(git rev-parse --abbrev-ref HEAD)" != "main" ]; then
+          git branch --track main origin/main
```

Comment:
shouldn't branch be a param?

Reply:
yes, it should.

Reply:
Resolved by adding fetch depth to the checkout step and will update docs.